ci/tests/vmachine/vmachine_test.py

```diff
@@ -0,0 +1,124 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.openvstorage.org/license
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ci.tests.general.general import test_config
+from ci.tests.general.connection import Connection
+from ci.tests.vpool import vpool_test
+from ci.tests.mgmtcenter import mgmt_center_test
+from ci.tests.general import general
+from ci import autotests
+import time
+
+testsToRun = general.get_tests_to_run(autotests.get_test_level())
+
+VPOOL_NAME = test_config.get('vpool', 'vpool_name')
+TEMPLATE_SERVERS = ['http://sso-qpackages-loch.cloudfounders.com/templates/openvstorage', 'http://172.20.3.8/templates/openvstorage']
+
+
+template_source_folder = '/fio_debian/'
+template_image = 'debian.qcow2'
+template_target_folder = '/var/tmp/templates/'
+
+NUMBER_OF_DISKS = 10
+GRID_IP = test_config.get('main', 'grid_ip')
+
+
+def download_template(server_location):
+    general.execute_command('wget -P {0} {1}{2}{3}'.format(template_target_folder, server_location, template_source_folder, template_image))
+    general.execute_command('chown root {0}{1}'.format(template_target_folder, template_image))
+
+
+def get_template_location_by_ip(ip):
+    if ip.split('.')[0] == '172' and ip.split('.')[1] == '20':
+        return TEMPLATE_SERVERS[1]
+    else:
+        return TEMPLATE_SERVERS[0]
+
+
+def check_template_exists():
+    out, err = general.execute_command('[ -d {0} ] && echo "Dir exists" || echo "Dir does not exists"'.format(template_target_folder))
+    if 'not' not in out:
+        general.execute_command('rm -rf {0}'.format(template_target_folder))
+        general.execute_command('mkdir {0}'.format(template_target_folder))
+    download_template(get_template_location_by_ip(GRID_IP))
+
+
+def setup():
+    check_template_exists()
+    vpool_test.setup()
+    mgmt_center_test.setup()
+    vpool_test.add_vpool()
+
+
+def teardown():
+    api = Connection.get_connection()
+    vpool_list = api.get_component_by_name('vpools', VPOOL_NAME)[0]
+    assert len(vpool_list), "No vpool found where one was expected"
+    vpool = vpool_list[0]
+    if vpool:
+        _, __ = general.execute_command("rm -rf /mnt/{0}/*.raw".format(VPOOL_NAME))
+        general.api_remove_vpool(VPOOL_NAME)
+    vpool_test.teardown()
+    # @todo change mgmgt teardown to delete recently added hmc
+    mgmt_center_test.teardown()
+
+
+def create_raw_vdisk_from_template(template_folder, image_name, vpool_name, disk_name):
+    _, __ = general.execute_command('qemu-img convert -O raw {0}{1} /mnt/{2}/{3}.raw'.format(template_folder, image_name, vpool_name, disk_name))
```

Comment:
I would like to have the output of these command logged to /var/log/ovs/autotests/vmachine.log
e.g. https://github.com/openvstorage/integrationtests/blob/master/ci/tests/backend/alba.py
from ci.tests.general.logHandler import LogHandler

logger = LogHandler.get('vmachine', name='fio')
logger.logger.propagate = False